mathinspector/console/interpreter.py

```diff
@@ -233,21 +257,39 @@ def _on_button_right(self, event):
 		if tag_ranges:
 			option.extend([{
 				"label": "Copy",
-				"command": lambda: self.clipboard_append(self.get(*tag_ranges))
+				"command": lambda: self.copy_to_clipboard(self.get(*tag_ranges))
 			}])
 
 		self.menu.show(event, option + [{
 			"label": "Clear Log",
 			"command": clear
 		}])
 
-
 	def _click(self, event, tag):
 		if not self.hover_range: return
 		content = self.get(*self.hover_range)
 		if tag == "error_file":
 			open_editor(os.path.abspath(content[1:-1]))
 
+class StdOut:
+	def __init__(self, write):
+		self.buf = []
+		self.write = write
+
+	def buffer(self, *args, **kwargs):
+		if len(args) == 0: return
+		self.buf.extend(args)
```

Comment:
this isn't at all like `sys.stdout.buffer` though...

Reply:
I tried to read up on this and then did my best to guess how it works based on my understanding of the word "buffer" and how I thought it worked.

Which part is wrong?

Reply:
well first up `sys.stdout` inherits from `TextIOWrapper`. There be a bushel of ye moste dark magike in mathinspector overloading builtin functionality but you can read up on io buffers here: https://docs.python.org/3/library/io.html

Reply:
new idea
```python
sys.stdout = StdOut(sys.stdin, self.write)
sys.stderr = StdOut(sys.stdin, self.write)

...

class StdOut(TextIOWrapper):
    def __init__(self, buffer, write, **kwargs):
        super(StdOut2, self).__init__(buffer, **kwargs)
```

It seems to work.  Before pushing this, I am wondering if you like this better or if it needs more work?
<img width="763" alt="Screen Shot 2021-02-25 at 12 34 14 PM" src="https://user-images.githubusercontent.com/60369559/109192899-cc6c8b80-7765-11eb-8f2c-998f8d708914.png">
        self.write = write

Reply:
`sys.stdin` are you sure? :)

Reply:
> sys.stdin are you sure? :)

lol, I am not sure about this at all, except in as much as it seems to work.  Some of the previous black magic didn't seem required in interpreter.py any longer, and I checked around to try and break stuff, it seems fine.  This is even darker black magic and I like it more :)